File: src/AcroStack/Services/Dtos/Chat/ChatMessageReactionDto.cs
using System;
using Volo.Abp.Application.Dtos;

namespace AcroStack.Services.Dtos.Chat;

/// <summary>A reaction (emoji) left by a user on a chat message.</summary>
public class ChatMessageReactionDto : EntityDto<Guid>
{
    /// <summary>IdentityUser.Id of the user who left the reaction.</summary>
    public Guid UserId { get; set; }

    /// <summary>User name of the user who left the reaction.</summary>
    public string UserName { get; set; } = string.Empty;

    /// <summary>The reaction emoji.</summary>
    public string Reaction { get; set; } = string.Empty;

    /// <summary>When the reaction was created.</summary>
    public DateTime CreationTime { get; set; }
}

File: react/src/api/models/acroStack/chat/ChatMessageReactionDto.ts
/* oxlint-disable */

/**
 * @description A reaction (emoji) left by a user on a chat message.
 * @type object
 */
export type AcroStackChatChatMessageReactionDto = {
  /**
   * @description
   * Format: `uuid`
   * @type string | undefined
   */
  id?: string;
  /**
   * @description IdentityUser.Id of the user who left the reaction.
   *
   * Format: `uuid`
   * @type string | undefined
   */
  userId?: string;
  /**
   * @description User name of the user who left the reaction.
   * @type string | undefined
   */
  userName?: string | null;
  /**
   * @description The reaction emoji.
   * @type string | undefined
   */
  reaction?: string | null;
  /**
   * @description When the reaction was created.
   *
   * Format: `date-time`
   * @type string | undefined
   */
  creationTime?: string;
};
